### PR TITLE
Implement keyword distribution assessment

### LINKED ISSUE #1545: Copy the keyword distribution assessment to the feature/synonyms branch
The keyword distribution assessment that has been developed within feature/recalibration now should be made available in YoastSEO.js before the release of feature/recalibration. Therefore, the assessment and the research should be copied to the feature/synonyms branch.

### PR DESCRIPTION
**Pull request by manuelaugustin**
## Summary

This PR can be summarized in the following changelog entry:

* Adds an assessment that checks whether there are any parts of the text (30% of the total or more) that don't include the keyword.

## Relevant technical choices:

* The keyword distance is based on the indices of the keywords, i.e. it is character-based (rather than word-based).
* The assessment gets triggered if the text contains 200 words or more and if there are at least 2 keyword occurrences in the text.

## Test instructions

This PR can be tested by following these steps:

See https://github.com/Yoast/YoastSEO.js/pull/1403

Fixes #1545

### CODE REVIEW

File: src/assessments/seo/largestKeywordDistanceAssessment.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		const defaultConfig = {
+			overRecommendedMaximumKeywordDistance: 40,
+			recommendedMaximumKeywordDistance: 30,
```
Comment (nataliashitova) [inline src/assessments/seo/largestKeywordDistanceAssessment.js]: I am debating with myself if lowing down these boundaries would be a smart move. On the one hand, all well-written texts I tested (a.k.a. Yoast Blog) pass this test with flying colors. On the other hand, with non-working morphology these boundaries might be a little too low for many amateur writers. Could you please give these boundaries a little more test?

File: src/assessments/seo/largestKeywordDistanceAssessment.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	 */
+	getResult( paper, researcher, i18n ) {
+		this._largestKeywordDistance = researcher.getResearch( "largestKeywordDistance" );
+		let assessmentResult = new AssessmentResult();
+
+		const calculatedResult = this.calculateResult( i18n );
+
+		assessmentResult.setScore( calculatedResult.score );
+		assessmentResult.setText( calculatedResult.resultText );
+		assessmentResult.setHasMarks( calculatedResult.score < 2 );
```
Comment (nataliashitova) [inline src/assessments/seo/largestKeywordDistanceAssessment.js]: Why not to allow markings for the OK case? I am afraid it is an artefact of the previously used scoring schema that only had good and bad.
Please also fix it in feature/recalibration.

Comment (manuelaugustin) [reply src/assessments/seo/largestKeywordDistanceAssessment.js]: Fixed for this PR

Concerning `feature/recalibration`: I think once we'll have merged `feature/synonyms` into `feature/development` and `feature/development` into `feature/recalibration` that will take care of it.

File: src/assessments/seo/largestKeywordDistanceAssessment.js (hunk 1 above)
Comment (manuelaugustin) [reply src/assessments/seo/largestKeywordDistanceAssessment.js]: I've raised the boundaries by 10%. So now you only get a bad score if half of your text doesn't contain the keyword, and an okay score if 40% of your text doesn't contain your keyword.

**Comment by nataliashitova** [comment]
CR and acceptance 👍

**Pull request merged**